### ISSUE TITLE
GitHub: Social Network Login Failure

### ISSUE THREAD

**Issue by akkisagiraju** [pennersr/django-allauth#3256]
Hello, I'm using django-allauth for the first time. Whenever I try to sign in with GitHub, I'm getting the below error:

`{'provider': 'github', 'code': 'unknown', 'exception': HTTPError('404 Client Error: Not Found for url: https://api.github.com/user/emails')}`

This is my configuration for the socialapp.
```
SOCIALACCOUNT_PROVIDERS = {
    "github": {
        "VERIFIED_EMAIL": True,
        'SCOPE': [
            'read:user',
            'repo',
            'read:org',
        ],
    },
}
```

I can confirm that the app is registered properly and I can see that the user has been added to the GitHub app.

**Comment by pennersr** [comment]
The provider indeed accesses https://api.github.com/user/emails -- but that is a valid URL:

https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user

Though, in the list of response status codes 404 is mentioned, without being clear on under which circumstances that can occur. Does the account you are using have an email address configured over at GitHub?

**Comment by akkisagiraju** [comment]
Hi @pennersr. Sorry I couldn't respond to your comment earlier.

> Does the account you are using have an email address configured over at GitHub?

Yes, the email associated with the account I'm using to sign up is the same email that is configured on GitHub.

**Comment by pennersr** [comment]
It's still unclear to me under what circumstances a 404 can occur, but given that it is documented in their documentation and that you are running into this issue, I think we need to adjust this bit:

```
    def get_email(self, headers):
        email = None
        resp = requests.get(self.emails_url, headers=headers)
        resp.raise_for_status()
```

to add a `if resp.status_code == 404: return None`. You might want to try changing that to see if you find any other issues in your flow.

**Comment by pennersr** [comment]
Closed via 763b9888

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3256
